programs/autocrat/src/state/amm.rs

```diff
@@ -21,6 +21,8 @@ pub struct Amm {
     pub total_ownership: u64,
     pub num_current_lps: u64,
 
+    // IMO we should just keep it as simple as possible and just do normal TWAP
```

Comment:
Hmm. I like simplicity when all things are otherwise equal. IMO a plain twap is definitely less good. If someone proposes something and no one notices for 12 hours, then that person effectively locks in a free 12 hours. Doing it weighted by liquidity allows for recovery. And from the non-security perspective I think its just a better representation of the market.

Reply:
I don't think it's a huge deal either way, I'll happily revert if my arguments don't persuade you.